EKF/ekf_helper.cpp

```diff
@@ -964,14 +964,127 @@ void Ekf::get_gyro_bias(float bias[3])
 	memcpy(bias, temp, 3 * sizeof(float));
 }
 
-// get the diagonal elements of the covariance matrix
+// covariance propagation from quaternions to euler angles using the covariance law
+// source: "Development of a Real-Time Attitude System Using a Quaternion
+// Parameterization and Non-Dedicated GPS Receivers", John B. Schleppe 1996
+void Ekf::propagate_covariances_from_quat_to_euler(matrix::SquareMatrix<float, 3> &euler_cov)
```

Comment:
You can return this by value.

Reply:
Yeah I wanted to keep the consistency here but since this is only consumed internally, I guess we can return by value.

Reply:
Yes, but you can still have all the new methods here returning values and marked const.

Reply:
@dagar done!